### DISCUSSION TITLE
Map group memberships from an OIDC claim.

### DISCUSSION THREAD

**Discussion by fostermi** [drakkan/sftpgo#1707] [Q&A]
Still working on the POC for this and I have another question. Is there a way to map an OIDC claim (called, "groups") to user groups in sftpgo? I'd rather not have to manage group membership in sftpgo, but instead on our OIDC federated backend (currently Keycloak, but will eventually be Azure AD).

**Comment by fostermi** [comment]
I was hoping it was as simple as putting the placeholder "{{IDPFieldgroup}}" that is available as described here, https://sftpgo.github.io/latest/eventmanager/  , but the software doesn't like that, throwing an error:
```
{"level":"error","time":"2024-07-26T23:23:55.044","sender":"eventmanager","message":"unable to handle IDP login event \"IDP login user\", err: json: cannot unmarshal string into Go struct field User.groups of type []sdk.GroupMapping"}
{"level":"debug","time":"2024-07-26T23:23:55.044","sender":"httpd","message":"unable to get the sftpgo user associated with oidc token: json: cannot unmarshal string into Go struct field User.groups of type []sdk.GroupMapping"}
```
The document says the field contains a `string`, but I don't know why that shouldn't be configureable based on the custom field.

**Reply by marsteel** [reply]
in keycloak for the client `sftpgo`, create a role called `admin`
in keycloak create a group

 `sftpgo-webadmin` and attach client role to this group in `groups details` - `Role mapping`

Clients -> Client details ->Dedicated scopes> sftpgo-dedicated, add mapper `client roles`
`Client ID` set to client id `sftpgo`
`Token Claim Name` set to `roles` which matches `SFTPGO_HTTPD__BINDINGS__OIDC__ROLE_FIELD`
`Claim JSON Type` set to `String`
`Add to ID token` set to `On`
